src/build/env.py

```diff
@@ -51,191 +24,166 @@ class IsolatedEnvironment(object):
     Non-standard paths injected directly to sys.path still be passed to the environment.
     """
 
-    _MANIPULATE_PATHS = ('purelib', 'platlib')
+    def __init__(self, path, executable, install_executable):
+        # type: (str, str, str) -> None
+        """
+        Define an isolated build environment.
 
-    def __init__(self, remove_paths, _executable=sys.executable):  # type: (Sequence[str], str) -> None
+        :param path: the path where the environment exists
+        :param executable: the python executable within the environment
+        :param install_executable: an executable that allows installing packages within the environment
         """
-        :param remove_paths: Import paths that should be removed from the environment
+        self._path = path
+        self._install_executable = install_executable
+        self._executable = executable
+
+    @classmethod
+    def for_current(cls):  # type: () -> IsolatedEnvironment
+        """
+        Create an isolated build environment into a temporary folder that matches the current interpreter.
+
+        :return: the isolated build environment
         """
-        self._env = {}  # type: Dict[str, Optional[str]]
-        self._env_vars = {}  # type: Dict[str, str]
-        self._path = None  # type: Optional[str]
-        self._remove_paths = []  # type: List[str]
-        self._executable = _executable
-        self._old_executable = None  # type: Optional[str]
-        self._pip_executable = None  # type: Optional[str]
```

Comment:
I've realized that ``pip_executable`` is assuming too much. In the case of `conda` for example, the user would be most likely to use `conda`. To acknowledge this I've renamed this to `instal_executable`.

Reply:
I would prefer to keep the original name. If you want to use conda you should subclass `IsolatedEnvironment`, where you can have a `_conda_executable` or something else.

Reply:
Well, potentially. It's a question of design. But then the base should be an abstract class because within conda you should not have pip.

Reply:
Would you like an abstract base class `IsolatedEnvironment` that defines the API as expected by the project builder, and then we add implementation `IsolatedEnvironmentVenvPip`? Because then yes naming it directly ``pip_executable`` makes a lot of sense.

Reply:
Why do we need this instead of `[self._executable, "-m", "pip"]`?

Reply:
Because there's no reason for us to have to provision pip inside the isolated environment, we can just use the pip install alongside the pip project and pass in ``--prefix`` to this path.

Reply:
Would `'sys.executable', '-m', 'pip'` not work in that case?

Reply:
Not in general, hence why is a config value. Sometimes the host doesn't have pip, and in such cases, it's good to fall back to ensurepip. Why you don't like it?

Reply:
I prefer `-m` as a rule since entry points add an additional layer of indirection that can easily go wrong in edge cases.

I also wanted to reduce the number of instance members since the more there are the more possibility things can go wrong, but from your explaination above, that’s no longer an issue now since there’s no way to avoid it.

Reply:
Not we're still using -m (if you read the install method implementation), the pip_executable is not actually the pip executable script, but the python executable having the pip inside 🤔 The name is not best, but it's an implementation detail (part of a private class).

Reply:
I see, thanks for the clarification.